New profile for busses

DanielAMCON opened the pull request:
My company needed to be able to calculate routes specifically for buses. As discussed in #2659 , I created a new profile for this purpose. We would like to make this available for others.

As already explained in the issue, the VehicleTagParser will also be removed in the long run. Therefore, I can understand if the PR will not be merged. If the case I described can now be implemented by other means, I will of course not merge the PR either. I am pleased about your feedback.

DanielAMCON commented:
Oh, that would be great. Thank you in advance!

DanielAMCON commented:
Thanks for commiting the changes. I changed the config file as you described and the bus profile was created successfully.

Unfortunately, the routing behaviour is not working as before. Here are some examples:

1. The bus should be able to take the road that is marked in red (https://www.openstreetmap.org/way/81032078) because there is a bus stop. Even if I set a waypoint there, it takes the long way.
![BusError1](https://user-images.githubusercontent.com/113018902/207540065-b738d982-39c3-4855-9d08-54c77698942b.png)
2. Here, the bus also cannot reach the bus stop. The road is https://www.openstreetmap.org/way/160155244.
![BusError2](https://user-images.githubusercontent.com/113018902/207540983-ec43f3d0-03c3-4478-9511-6f464a4337ab.png)
3. This bus stop cannot be reached either. The road is https://www.openstreetmap.org/way/30016603.
![BusError3](https://user-images.githubusercontent.com/113018902/207541295-03786c71-4105-446a-b27e-266778fffd4e.png)

There are no barriers in all of that cases. I assume that this is related to the _motorvehcile_, _motorcar_, and _vehicle_ tags, as they are each set to _no_ for the roads mentioned. Could you please take a closer look at this?

DanielAMCON commented:
Yes, I've got time for this. I'll give it a try!

easbar commented:
But doesn't `psv=yes` rather belong to a to-be-created transportation mode dependent access value? We aren't going to create separate encoded values for `hgv=yes`, `motorcycle=yes`, `foot=yes` etc., are we? Also adding a psv encoded value won't change turn restrictions that can also be transportation mode specific. So yes, adding a psv encoded value might improve the situation for this specific bus profile, but in the longer run we need a proper replacement (and strategy) for what is currently still done in the vehicle tag parsers.

easbar commented:
> Hm, you mean we rather need a boolean encoded value per transportation mode?

Yes, so far I thought we were aiming for a boolean value to represent access per transportation mode. This would be a compound of multiple different OSM tags (for example oneway, different barriers etc.) including the access tags. But of course there are many options this can be dealt with. For example would you like a separate oneway encoded value instead of putting it into the access encoded value? And if yes, will there even be an access encoded value?

> This should be handled from the roads vehicle via graph.vehicles: roads|transportation_mode=BUS,car

Yes, currently `graph.vehicles` is the place where this can be configured. The only problem here is that so far we do not support `transportation_mode=bus`, but only `transportation_mode=psv`. But even when we use `transportation_mode=psv` we will currently miss those restrictions that are tagged as `bus=yes`.

DanielAMCON commented:
So, I now added psv as a new EncodedValue. At first, it didn't work. After analyzing this a bit further, I realized that it works when I delete this part of the bus.json:
```
{
  if": "true",
  "limit_to": "car_average_speed * 0.9"
}
```
So it now looks like this:
```
{
  "distance_influence": 90,
  "speed": [
    {
      "if": "true",
      "limit_to": "120"
    }
  ],
  "priority": [
    {
      "if": "car_access == false && psv != YES || max_width < 3 || max_height < 4",
      "multiply_by": "0"
    }
  ]
}
```
I am very confused now. Is `car_average_speed` not working here? And if yes, why exactly doesn't it work? Do you guys have an idea?

Regarding your discussion, I implemented this using an EncodedValue. If you give me further information, I could also implement this using a boolean. But based on your dicussion I actually don't quite know how to do this.
